### ISSUE TITLE
Symlinks replaced by applications (for Library/Preferences)

### ISSUE THREAD

**Issue by jonasdiemer** [lra/mackup#1854]
Hi,

I noticed that applications like BetterTouchTool and iTerm2 replace the symlinks created by mackup in `~/Library/Preferences/` with physical copies of the config files.

Is this expected? Would `mackup restore` still allow me to sync preferences to a device? 

Regards
Jonas

**Comment by azuenko** [comment]
For anyone facing the same issue with iTerm2, a quick workaround would be to say iTerm2 to store the configuration in a custom folder, e.g. `~/.config/iterm2`. Then, use a custom `iterm2-ext-config.cfg` :

```
[application]
name = iTerm2-ext-config

[xdg_configuration_files]
iterm2
```

Unfortunately, this approach works only for iTerm, and if an app keeps replacing a symbolic link with a file in, and there's no way to isolate that file in a separate folder, I'm afraid there's nothing can be done here.

**Comment by jonasdiemer** [comment]
Found some discussions on how to approach it here: https://apple.stackexchange.com/questions/385102/how-to-properly-symlink-files-in-the-library-preferences

One option might be to lock the file (`chflags nouchg`).

**Comment by jonasdiemer** [comment]
This behavior has nothing to do with the storage destination (iCloud or whatever), as the symlink that is overwritten lies on the "source storage".

**Comment by jonasdiemer** [comment]
No, the symlinks that are put in place of the plist files have nothing to do with the storage destination - they simply point to another path in the filesystem (AFAIK).

**Comment by lra** [comment]
> Is it fair to assume this might lead to Mackup being obsolete?

If every application and OS decide to throw away the POSIX concept of file linking, yes. But before it happens, the strategy has always been to remove support for those impacted applications in Mackup.

**Comment by anthonysomerset** [comment]
This appears to me to be specific to MacOS Ventura (13) - I don't appear to observe this issue on older releases

**Comment by Morriz** [comment]
> This appears to me to be specific to MacOS Ventura (13) - I don't appear to observe this issue on older releases

If it would be as easy as adding that `chflags nouchg` operation, then I hope it will be added soon :)

**Comment by Triloworld** [comment]
Owner of Raycast start to remove sync options without anybody notice because of this issue: https://github.com/lra/mackup/issues/1980#issuecomment-1884729862
@gridsystem have right. Applications may have they own solution that block copy operation but it seams that is system issue
